Modules/LearningSequence/classes/Player/class.ilLSStateDB.php

```diff
@@ -191,6 +191,9 @@ public function deleteForItem(int $lso_ref_id, int $item_ref_id)
 		foreach ($all_states as $usr_id=>$state_entry) {
 			list($current_item, $states) = $state_entry;
 
+			if (!is_array($states)) {
```

Comment:
This seems to be missplaced here? Why isn't this an array in the first place? I think this rather belongs into `buildStates`.